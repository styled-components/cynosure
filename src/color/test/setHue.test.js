// @flow
import setHue from '../setHue'

describe('setHue', () => {
  it('should update the hue and return a hex color', () => {
    expect(setHue(42, '#CCCD64')).toMatchSnapshot()
  })

  it('should update the hue of an 8-digit hex color and return a hex color', () => {
    expect(setHue(244, '#6564CDB3')).toMatchSnapshot()
  })

  it('should update the hue and return a color with opacity', () => {
    expect(setHue(244, 'rgba(101,100,205,0.7)')).toMatchSnapshot()
  })

  it('should update the hue when passed a string for hue', () => {
    expect(setHue('42', '#CCCD64')).toMatchSnapshot()
  })
})
