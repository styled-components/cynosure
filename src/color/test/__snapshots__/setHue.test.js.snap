// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setHue should update the hue and return a color with opacity 1`] = `"rgba(107,100,205,0.7)"`;

exports[`setHue should update the hue and return a hex color 1`] = `"#cdae64"`;

exports[`setHue should update the hue when passed a string for hue 1`] = `"#cdae64"`;
